src/pages/CircleAdminPage/HedgeyIntegrationSettings.tsx

```diff
@@ -0,0 +1,186 @@
+import { useEffect, useState } from 'react';
+
+import {
+  createCircleIntegration,
+  updateCircleIntegration,
+} from 'lib/gql/mutations';
+
+import { Button, Flex, Modal, Select, SelectOption, Text } from 'ui';
+
+function HedgeyIntro() {
+  return (
+    <>
+      <Text h3 semibold css={{ mb: '$sm' }}>
+        Hedgey Integration
+      </Text>
+      <Text p size="small" css={{ mb: '$md' }}>
+        Hedgey is a protocol for locking tokens during fundraising and payroll.{' '}
+        <a href="https://hedgey.finance" target="_blank" rel="noreferrer">
+          Find out more
+        </a>
+      </Text>
+    </>
+  );
+}
+
+export const hedgeyLockPeriods: SelectOption[] = [
+  { label: '1 month', value: '1' },
+  { label: '2 months', value: '2' },
+  { label: '3 months', value: '3' },
+  { label: '4 months', value: '4' },
+  { label: '5 months', value: '5' },
+  { label: '6 months', value: '6' },
+  { label: '12 months', value: '12' },
+  { label: '18 months', value: '18' },
+  { label: '24 months', value: '24' },
+  { label: '48 months', value: '48' },
+];
+
+export default function HedgeyIntegrationSettings(props: {
+  circleId: number;
+  integration: {
+    id: number;
+    type: string;
+    data: { enabled: boolean; lockPeriod: string; transferable: string };
+  };
+}) {
+  const [circleIntegrationId, setCircleIntegrationId] = useState(
+    undefined as number | undefined
+  );
+  const [hedgeyEnabled, setHedgeyEnabled] = useState(false);
+  const [hedgeyLockPeriod, setHedgeyLockPeriod] = useState('12');
+  const [hedgeyTransferable, setHedgeyTransferable] = useState('0');
+  const [showDisableModal, setShowDisableModal] = useState(false);
+
+  useEffect(() => {
+    setCircleIntegrationId(props.integration?.id);
+    setHedgeyEnabled(props.integration?.data.enabled || false);
+    setHedgeyLockPeriod(props.integration?.data.lockPeriod || '12');
+    setHedgeyTransferable(props.integration?.data.transferable || '0');
+  }, [props]);
+
+  const onSaveHedgeyIntegration = async (e: any) => {
```

Comment:
i was editing this file just to simplify the rendering for the `!hedgeyEnabled` case a bit, but i ran into trouble when i was testing my changes. i think there are a few bugs here:

- this method (onSaveHedgeyIntegration) should be called when the "Enable Hedgey Integration" button is clicked
- `enabled` should always be `true` -- this isn't called to disable hedgey, right?

Reply:
That is correct, it is always going to be true.